### PR TITLE
fix(adapter-pg/pg-worker/neon): pass customParsers to the query

### PR DESCRIPTION
**Pull request by Jolg42**
Related to https://github.com/prisma/prisma/issues/23505

### CODE REVIEW

**Comment by Jolg42** [comment]
TODO: check adapter-neon (& others)

File: packages/adapter-pg/src/pg.ts [hunk 1]
```diff
@@ -87,7 +89,31 @@ class PgQueryable<ClientT extends StdClient | TransactionClient> implements Quer
...
 
     try {
-      const result = await this.client.query({ text: sql, values: fixArrayBufferValues(values), rowMode: 'array' })
+      const result = await this.client.query(
+        {
+          text: sql,
+          values: fixArrayBufferValues(values),
+          rowMode: 'array',
+          types: {
+            // Error: packages/adapter-pg build: src/pg.ts(98,13): error TS2769: No overload matches this call.
```
Comment (SevInf) [inline packages/adapter-pg/src/pg.ts:98]: Would adding explicit type annotation to `customParsers` fix it?
`Record<number, types.TypeParser<string, unknown>>` should be the correct type.
Alternatively, maybe `as const` after `customParsers` declaration?

Comment (SevInf) [reply packages/adapter-pg/src/pg.ts:98]: Ah, I see the issue. Apparantly, `pg-types` types expect you to handle both binary and text protocol where versions, where as far as i can see, `pg` will ever pass only text version. We can put that into the comment instead, and then add assertion that `format === 'text'` before reaching out for our parsers.

Comment (Jolg42) [reply packages/adapter-pg/src/pg.ts]: I did the update

**Review by SevInf — APPROVED** [review]

**Pull request merged**